generate-github-io

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+cdir=`pwd`
+branch=`git rev-parse --abbrev-ref HEAD`
+clone=~/trydent-temp-dir/TrydentEngine
+archive=~/trydent-temp-dir/temp-trydent-javadoc.tar
+
+mkdir ~/trydent-temp-dir
+
+# This is here so that if something goes wrong, we don't erase
+# anything important.
+mkdir safety-dir
+cd safety-dir
+
+git clone https://github.com/AaronFriesen/Trydent.git $clone
+cd $clone
+git checkout $branch
+gradle clean
+gradle javadoc
+cd build/docs/javadoc
+tar -cvf $archive *
+cd $clone
+git checkout gh-pages
+rm -rf "$clone/*"
+git ls-files --deleted | xargs git rm
+tar -xvf $archive 
+tar -tf $archive | xargs git add
+rm "$archive"
+git commit -am "Updating javadocs for github.io."
+git push origin gh-pages
```

Comment:
Hmm, we could change it to only checkout the master branch's javadocs, but since javadoc is broken on master currently I think I'd rather leave the script as-is for now, merge in the javadoc fixes, then circle back and update the script in a new branch.